### PR TITLE
Created Car data and License plate checking API calls

### PR DESCRIPTION
**Pull request by aurisgo**

### CODE REVIEW

File: EPolicija/EPolicija.csproj [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (modestas-jucevicius) [inline EPolicija/EPolicija.csproj]: Why create new projects, this complicates compiling, and needlessly creates multiple files

File: Models/Cars/Car.cs [hunk 2]
```diff
@@ -1,51 +1,22 @@
...
+        [Required]
         public int UserId { get; set; }
-        public string LicensePlate { get; set; }    // automobilio registracijos numeris
-        public string ColorName { get; set; }       // spalva
-        public string MakeName { get; set; }        // gamintojo pavadinimas
-        public string Model { get; set; }           // modelis
-        public string BodyType { get; set; }        // tipas
-        public string Year { get; set; }            // pagaminimo metai
+        [Required]
+        public string LicensePlate { get; set; }
```
Comment (modestas-jucevicius) [inline Models/Cars/Car.cs:12]: Why remove comments?

File: Models/Cars/CarFactory.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                IsReported = car.IsReported,
+                Image = ImageFactory.GetInstance().CreateImage(image)
+            };
+        }
+
+        public Car CreateCar(int id, int userId, string licensePlate, string colorName,
+                            string makeName, string model, string bodyType, string year,
+                            CarStatus status, bool isReported)
+        {
+            return new Car
```
Comment (modestas-jucevicius) [inline Models/Cars/CarFactory.cs:44]: Car and capturedCar seems almost identical, maybe make image an optional parameter and use 1 class, another solution would be inheritance

File: Models/Images/Image.cs [hunk 4]
```diff
@@ -1,35 +1,16 @@
...
 {
     public class Image
     {
+        [Required]
         public int Id { get; set; }
+        [Required]
         public int CarId { get; set; }
+        [Required]
         public long Timestamp { get; set; }
         public string Path { get; set; }
```
Comment (modestas-jucevicius) [inline Models/Images/Image.cs:12]: Storing image as bytes in system is a very bad approach, leave Path only and make it required

File: Models/Images/ImageFactory.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                instance = new ImageFactory();
+            }
+            return instance;
+        }
+
+        private ImageFactory()
+        {
+        }
+
+        public Image CreateImage(Image image)
```
Comment (modestas-jucevicius) [inline Models/Images/ImageFactory.cs]: This seems more like a clone instead of createImage, maybe rename?

File: Score/Evaluation.cs [hunk 6]
```diff
@@ -30,33 +30,34 @@ public static Evaluation Instance
         }
 
 
-        public int Evaluate(Car car)                     // Įvertinti automobilį taškais
+        public int Evaluate(User user, CarStatus status)
```
Comment (modestas-jucevicius) [inline Score/Evaluation.cs]: These changes seem redundant

Leftover bad code = returning int as error code

File: Services/Recognition/OpenALPRParser.cs [hunk 7]
```diff
@@ -53,7 +53,15 @@ public Car ParseUnit(JObject result)
             JObject modelObj = (JObject)models[0];
             string model = (string)modelObj["name"];
 
-            return new Car(plate, colorName, makeName, model, bodyType, year);
+            return new Car
```
Comment (modestas-jucevicius) [inline Services/Recognition/OpenALPRParser.cs:56]: Use your new factory

File: Storage/Data/CapturedCarService.cs [hunk 8]
```diff
@@ -22,7 +22,7 @@ public List<CapturedCar> FindByPlate(string licensePlate)
                            join image in images on car.Id equals image.CarId
                            where car.LicensePlate.Equals(licensePlate)
                            orderby image.Timestamp descending
-                           select new CapturedCar(car, image);
+                           select CarFactory.GetInstance().CreateCapturedCar(car, image);
```
Comment (modestas-jucevicius) [inline Storage/Data/CapturedCarService.cs:25]: Get instance once and reuse it

File: Storage/Data/CapturedCarService.cs [hunk 9]
```diff
@@ -61,16 +61,19 @@ public void Add(CapturedCar car)
...
-                                      select image;
+                IEnumerable<Image> imgs = from image in images
+                                          where car.Id == image.CarId
+                                          select image;
 
-            carData.Remove(car.Id);
-            foreach (var img in imgs)
-            {
-                imageData.Remove(img.Id);
+                carData.Remove(car.Id);
```
Comment (modestas-jucevicius) [inline Storage/Data/CapturedCarService.cs:72]: Seems like these find and remove, should be implemented into imageData, single responsibility

File: WebService/Cars/CapturedCarParser.cs [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+            Image img = new Image
+            {
+                Id = (int)image["id"],
+                CarId = (int)image["carId"],
+                Timestamp = (long)image["timestamp"],
+                Picture = Encoding.ASCII.GetBytes((string)image["picture"])
+            };
+
+            return new CapturedCar
+            {
```
Comment (modestas-jucevicius) [inline WebService/Cars/CapturedCarParser.cs]: Use factory

File: WindowsFormApp/Presenters/MainPresenter.cs [hunk 11]
```diff
@@ -23,10 +20,10 @@ class MainPresenter
         private OpenALPRRecognizer recognizer = new OpenALPRRecognizer();
         public Label scoreLabel;
         private ICarParser parser = new OpenALPRParser();
-        private readonly ICarStatusRequester requester = EPolicijaAPIRequester.GetInstance();
+        //private readonly ICarStatusRequester requester = EPolicijaAPIRequester.GetInstance();
```
Comment (modestas-jucevicius) [inline WindowsFormApp/Presenters/MainPresenter.cs:23]: Remove if unneeded instead of commenting

**Review by modestas-jucevicius — CHANGES_REQUESTED** [review]

File: EPolicija/EPolicija.csproj (hunk 1 above)
Comment (aurisgo) [reply EPolicija/EPolicija.csproj]: I'll wrap projects

**Pull request merged**